### PR TITLE
[Deselecting Chapters] Add feature flag

### PR DESCRIPTION
**Pull request by leandroalonso**
| 📘 Part of: #1396 |
|:---:|

Adds the remote feature flag for the upcoming deselecting chapters feature.

## To test

1. Go to `AppDelegate` and remove the `#if !DEBUG` inside `updateRemoteFeatureFlags`
2. On line `291` change `2.hour` to `10.seconds`
3. Run the app
4. ✅ Go to Profile > Settings > Beta Features > `deselectChapters` should be toggled off
5. Go to Firebase Console and change `deselect_chapters` to `true`
6. Re-run the app
7. ✅ Go to Profile > Settings > Beta Features > `deselectChapters` should be toggled on
8. Go to Firebase Console and change `deselect_chapters` to `Use in-app default`
6. Re-run the app
7. ✅ Go to Profile > Settings > Beta Features > `deselectChapters` should be toggled off

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Review by emilylaguna — CHANGES_REQUESTED** [review]
Can we make this a remote feature flag?

**Comment by leandroalonso** [comment]
@emilylaguna updated! I also did a small change so using remote feature flags is way easier moving forward.

**Review by emilylaguna — APPROVED** [review]
Working for me!

File: podcasts/AppDelegate.swift [hunk 1]
```diff
@@ -282,7 +282,11 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
             Constants.RemoteParams.addMissingEpisodes: NSNumber(value: Constants.RemoteParams.addMissingEpisodesDefault),
             Constants.RemoteParams.newPlayerTransition: NSNumber(value: Constants.RemoteParams.newPlayerTransitionDefault),
             Constants.RemoteParams.errorLogoutHandling: NSNumber(value: Constants.RemoteParams.errorLogoutHandlingDefault)
-        ])
+        ]
+        FeatureFlag.allCases.filter { $0.remoteKey != nil }.forEach { flag in
+            remoteConfigDefaults[flag.remoteKey!] = NSNumber(value: flag.default)
+        }
```
Comment (emilylaguna) [inline podcasts/AppDelegate.swift:288]: I'm not a huge fan of the force unwrapping here, but I don't have a better idea at the moment.

**Pull request merged**